app/code/Magento/Widget/Model/Widget.php

```diff
@@ -296,42 +308,66 @@ public function getWidgetsArray($filters = [])
      */
     public function getWidgetDeclaration($type, $params = [], $asIs = true)
     {
-        $directive = '{{widget type="' . $type . '"';
         $widget = $this->getConfigAsObject($type);
 
+        $directiveParams = '';
         foreach ($params as $name => $value) {
             // Retrieve default option value if pre-configured
-            if ($name == 'conditions') {
-                $name = 'conditions_encoded';
-                $value = $this->conditionsHelper->encode($value);
-            } elseif (is_array($value)) {
-                $value = implode(',', $value);
-            } elseif (trim($value) == '') {
-                $parameters = $widget->getParameters();
-                if (isset($parameters[$name]) && is_object($parameters[$name])) {
-                    $value = $parameters[$name]->getValue();
-                }
-            }
-            if (isset($value)) {
-                $directive .= sprintf(' %s="%s"', $name, $this->escaper->escapeHtmlAttr($value, false));
-            }
+            $directiveParams .= $value === null ? '' : $this->getDirectiveParam($widget, $name, $value);
```

Comment:
To avoid the ternary operator here, wouldn't be better to use an array_filter before the foreach? Like this:
```
array_filter($params, function($value) {
    return $value !== null;
})
```
It would be one step more before the foreach but since here we are processing just one widget I guess the performance impact is almost none and it would improve readability. What do you think?